geoindicators/src/main/groovy/org/orbisgis/RsuIndicators.groovy

```diff
@@ -619,4 +615,164 @@ static IProcess rsuEffectiveTerrainRoughnessHeight() {
+                        ifZindex = ", b.$zindex AS zindex"
+                        filtering = ""
+                        levelConsiderated.each { lev ->
+                            filtering += "$baseFiltering AND b.$zindex=$lev OR "
+                        }
+                        filtering = filtering[0..-4]
+                    }
+                    def selectionQuery = "DROP TABLE IF EXISTS $outputTableName; DROP TABLE IF EXISTS $roadInter; " +
+                            "CREATE INDEX IF NOT EXISTS ids_r ON $roadTable($geometricColumnRoad) USING RTREE; " +
+                            "CREATE INDEX IF NOT EXISTS ids_u ON $rsuTable($geometricColumnRsu) USING RTREE; " +
+                            "CREATE TABLE $roadInter AS SELECT a.$idColumnRsu AS id_rsu, " +
+                            "ST_AREA(a.$geometricColumnRsu) AS rsu_area, ST_INTERSECTION(a.$geometricColumnRsu, " +
+                            "b.$geometricColumnRoad) AS the_geom $ifZindex FROM $rsuTable a, $roadTable b " +
+                            "WHERE $filtering;"
+                    datasource.execute(selectionQuery.toString())
+
+                    // If all roads are considered at the same level...
+                    if(levelConsiderated==null){
+                        nameDens.add("rsu_linear_road_density")
+                        caseQueryDens = "SUM(ST_LENGTH(the_geom))/rsu_area AS rsu_linear_road_density "
+                        for (int d=angleRangeSize; d<=180; d+=angleRangeSize){
+                            caseQueryDistrib += "SUM(CASEWHEN(azimuth>=${d-angleRangeSize} AND azimuth<$d, length, 0)) AS " +
+                                    "rsu_road_direction_distribution_d${d-angleRangeSize}_$d,"
+                        }
+                    }
+                    // If only certain levels are considered independantly
+                    else{
+                        ifZindex = ", zindex "
+                        levelConsiderated.each { lev ->
+                            caseQueryDens += "SUM(CASEWHEN(zindex = $lev, ST_LENGTH(the_geom), 0))/rsu_area " +
+                                    "AS rsu_linear_road_density_h${lev.toString().replaceAll('-','minus')},"
+                            nameDens.add("rsu_linear_road_density_h${lev.toString().replaceAll('-','minus')}")
+                            for (int d=angleRangeSize; d<=180; d+=angleRangeSize){
+                                caseQueryDistrib += "SUM(CASEWHEN(azimuth>=${d-angleRangeSize} AND azimuth<$d AND " +
+                                        "zindex = $lev, length, 0)) AS " +
+                                        "rsu_road_direction_distribution_h${lev.toString().replaceAll("-","minus")}" +
+                                        "_d${d-angleRangeSize}_$d,"
+                            }
+                        }
+                    }
+
+                    // QUERIES EXECUTED DEPENDING ON THE CONTENT OF THE OPERATION LIST
```

Comment:
Upper case ?

Reply:
It was to differentiate to main steps in the script. I have transformed to lowercase and put 1. and 2.